File: 2020-CompPhys/Homework_solutions/HW1_2/problem 2/problem2.m
disp("integer test:  ")
x=1;
for iter=1:1024
    x=x*2;
    if(iter>1022)
        disp(strcat("2^",num2str(iter)," = ",num2str(x)))
    end
end
%returns an inf for the integer rep

disp("double test:  ")
y=1;
for iter=1:1024
    y=y*2;
    if(iter>1022)
        disp(strcat("2^",num2str(iter)," = ",num2str(y)))
    end
end
%returns an inf for the dlouble rep

File: 2020-CompPhys/Homework_solutions/HW1_2/problem 2/problem2.cpp
#include <iostream>
#include <cmath>


int main(){
	std::cout<<"integer test:  \n";
	int x=1;
	for(int iter=0;iter<32;iter++){
		x=x*2;
		if(iter>28){
			std::cout<<"2^"<<iter<<" = "<<x<<"\n";
		}
	}
	//becomes negative, then zero.
	
	std::cout<<"double test \n";
	double y=1.0;
	for(int iter=0;iter<1025;iter++){
		y=y*2.0;
		if(iter>1021){
			std::cout<<"2^"<<iter+1<<" = "<<y<<"\n";
		}
	}
	//returns inf.

	
	return 0;
}
